File: problems/0061/solution.py
from typing import Optional


# Definition for singly-linked list.
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next


class Solution:
    def rotateRight(self, head: Optional[ListNode], k: int) -> Optional[ListNode]:
        if head == None:
            return None
        array = []
        ptr = head
        while ptr != None:
            array.append(ptr)
            ptr = ptr.next
        m = len(array)
        if k == 0 or k % m == 0:
            return head
        k %= m
        array[m - 1].next = array[0]
        array[m - k - 1].next = None
        return array[m - k]

File: problems/0061/solution.cpp
#include <iostream>
#include <vector>
using namespace std;


struct ListNode {
    int val;
    ListNode *next;
    ListNode() : val(0), next(nullptr) {}
    ListNode(int x) : val(x), next(nullptr) {}
    ListNode(int x, ListNode *next) : val(x), next(next) {}
};

class Solution {
public:
    ListNode* rotateRight(ListNode* head, int k) {
        if (head == nullptr) return nullptr;
        vector<ListNode*> list;
        ListNode * ptr = head;
        while(ptr != nullptr) {
            list.push_back(ptr);
            ptr = ptr->next;
        }
        int m = list.size();
        if (k == 0 || k % m == 0) return head;
        k %= m;
        list[m - 1]->next = list[0];
        list[m - k - 1]->next = nullptr;
        return list[m - k];
    }
};

int main() {
    Solution sol;
    ListNode * head = new ListNode(0, new ListNode(1, new ListNode(2)));
    ListNode * newHead = sol.rotateRight(head, 4);
    while (newHead != nullptr){
        cout << newHead->val << ',' ;
        newHead = newHead->next;
    }
}